src/components/ToDo/ToDo.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { TodoMethod } from '../../context/ToDosContext';
+import { TodoType } from '../../types/types';
+
+type Props = {
+  todo: TodoType;
+};
+
+export const Todo: React.FC<Props> = ({ todo }) => {
+  const methods = useContext(TodoMethod);
+  const [changer, setChanger] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  const fieldRef = useRef<HTMLInputElement>(null);
+
+  const saveChange = () => {
+    const trimmed = title.trim();
+
+    if (trimmed) {
+      methods.renameTodo(todo.id, trimmed);
+    } else {
+      methods.deleteTodo(todo.id);
+    }
+
+    setChanger(false);
+  };
```

Comment:
Could be wrapped in a `useCallback` for improved performance

Reply:
Done!